internal/controllers/machine/machine_controller_phases_test.go

```diff
@@ -769,55 +747,40 @@ func TestReconcileBootstrap(t *testing.T) {
 					"dataSecretName": "secret-data",
 				},
 			},
-			machine: &clusterv1.Machine{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "bootstrap-test-existing",
-					Namespace: metav1.NamespaceDefault,
-				},
-				Spec: clusterv1.MachineSpec{
-					Bootstrap: clusterv1.Bootstrap{
-						ConfigRef: &corev1.ObjectReference{
-							APIVersion: "bootstrap.cluster.x-k8s.io/v1beta1",
-							Kind:       "GenericBootstrapConfig",
-							Name:       "bootstrap-config1",
-						},
-						DataSecretName: ptr.To("secret-data"),
-					},
-				},
-				Status: clusterv1.MachineStatus{
-					BootstrapReady: true,
-				},
-			},
-			expectResult: ctrl.Result{},
-			expectError:  false,
+			bootstrapConfigGetError: nil,
+			expectResult:            ctrl.Result{},
+			expectError:             false,
 			expected: func(g *WithT, m *clusterv1.Machine) {
 				g.Expect(m.Status.BootstrapReady).To(BeTrue())
-				g.Expect(*m.Spec.Bootstrap.DataSecretName).To(BeEquivalentTo("secret-data"))
+				g.Expect(m.Spec.Bootstrap.DataSecretName).NotTo(BeNil())
+				g.Expect(*m.Spec.Bootstrap.DataSecretName).To(ContainSubstring("secret-data"))
 			},
 		},
 		{
-			name: "existing machine, bootstrap provider is not ready, and ownerref updated",
+			name:    "bootstrap config ready with no bootstrap secret",
+			machine: defaultMachine.DeepCopy(),
 			bootstrapConfig: map[string]interface{}{
 				"kind":       "GenericBootstrapConfig",
 				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
 				"metadata": map[string]interface{}{
 					"name":      "bootstrap-config1",
 					"namespace": metav1.NamespaceDefault,
-					"ownerReferences": []interface{}{
-						map[string]interface{}{
-							"apiVersion": clusterv1.GroupVersion.String(),
-							"kind":       "MachineSet",
-							"name":       "ms",
-							"uid":        "1",
-							"controller": true,
-						},
-					},
 				},
 				"spec": map[string]interface{}{},
 				"status": map[string]interface{}{
-					"ready": false,
+					"ready": true,
 				},
 			},
+			bootstrapConfigGetError: nil,
+			expectResult:            ctrl.Result{},
+			expectError:             true,
+			expected: func(g *WithT, m *clusterv1.Machine) {
+				g.Expect(m.Status.BootstrapReady).To(BeFalse())
+				g.Expect(m.Spec.Bootstrap.DataSecretName).To(BeNil())
+			},
+		},
+		{
+			name: "bootstrap data secret and bootstrap ready should not change after bootstrap config is set",
```

Comment:
Is this correct. If yes, I will surface this in the contract

Reply:
We can probably tackle this separately, ideally the bootstrap data is deleted and the secret goes away once the machine is ready

Reply:
ACK, let's tackle this separately
(btw. I agreed we can tolerate the bootstrap secret being deleted when the machine is ready, I'm so so on allowing the bootstrap data secret name to change or be deleted)

Reply:
Probably it shouldn't be possible to change it. It's hard to say what the expected result is (during bootstrap, which one would be used? after bootstrap, the new value would never be used)